### DISCUSSION TITLE
In Budibase, I'm trying to add in-page html anchors to facilitate vertical navigation in a web page with many reports

### DISCUSSION THREAD

**Discussion by mmays77** [Budibase/budibase#12845] [Help] [answered]
I'm a new Budibase user running version 2.13.50 under Ubuntu 22.04 and docker compose in VirtualBox.

I have a reports page in Budibase that has several reports in a vertical layout. I'd like to have a set of nav buttons at the top that take users to the reports desired, and "top" buttons near each report that bring the user back to the buttons menu at the top of the page. 

I tried using embed commands to put the into the HTML, such as  
&lt;h2 id="PaymentDetailbyAccount"&gt;Payment Detail by Account&lt;/h2&gt; 

I then put in a button at the top that has a navigation action to go to screen /paymentreports/#PaymentDetailbyAccount. I also tried setting the nav to screen /#PaymentDetailbyAccount and #PaymentDetailbyAccount with no success.

All of these attempts to link to the anchor in the same page take me to the default application screen. My "#top" link buttons do the same thing. I haven't been able to find any documentation or information on page navigation of this nature.

There must be a simple way to do vertical page navigation like this, but I'm not finding the secret. I know I can put every report on a separate page, but that would mean replicating and maintaining all the filters and options pickers for each page separately.

Any suggestions would be appreciated. Thanks in advance!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @mmays77,

Currently I don't believe there is a way to do this natively within Budibase. You might be able to do some hack with form fields as these have an action called [scroll to field.](https://docs.budibase.com/docs/form-actions#scroll-to-field) Additionally if you believe this is functionality that should be added to Budibase feel free to create a feature request [here](https://github.com/Budibase/budibase/issues/new/choose).

The navigate to component is really only used for navigating internally or externally. There is likely some logic that prevents add ons from being attached to the internal URLs. (As i can see it being stripped whenever attempting this.) This means that a core change would need to happen to allow for the functionality you are seeking with embed components. Additionally, components within Budibase don't allow for set ids, this means that even if the above change was done you wouldn't be able to specifically target a component without this change being made as well.

I don't think putting this on a separate page would be as difficult. It wouldn't require any duplication in my opinion, as you would just add filters to the data provider to get all relevant reports based on category/user filters that you add. This could all be done a single page to provide that dynamic generation feel.

**Comment by mmays77** [comment]
Thank you for your suggestions! 

The Form Step approach was perfect for what I wanted, and turned out to be way cooler than what I was trying. The Form Step sections for each report are only visible 1 at a time, so the reports page looks very modern compared to the long, 1990's style page that I was thinking about. I was able to hide dummy scroll-to fields in multiple field groups using conditional UI, and it works great.

Thanks again!
